src/System.Device.Gpio.Tests/GpioControllerTestBase.cs

```diff
@@ -130,30 +131,203 @@ public void OpenPinDefaultsModeToInput()
         [Fact]
         public void AddCallbackTest()
         {
-            while (!Debugger.IsAttached)
-                Thread.Sleep(1000);
-            Debugger.Break();
-
-            ManualResetEvent mre = new ManualResetEvent(false);
             bool wasCalled = false;
             using (GpioController controller = new GpioController(GetTestNumberingScheme(), GetTestDriver()))
             {
                 controller.OpenPin(InputPin, PinMode.Input);
                 controller.OpenPin(OutputPin, PinMode.Output);
                 controller.RegisterCallbackForPinValueChangedEvent(InputPin, PinEventTypes.Rising, callback);
                 controller.Write(OutputPin, PinValue.High);
-                mre.WaitOne(TimeSpan.FromSeconds(5));
+                Thread.Sleep(10);
                 Assert.True(wasCalled);
             }
 
             void callback(object sender, PinValueChangedEventArgs pinValueChangedEventArgs)
             {
                 wasCalled = true;
-                mre.Set();
+            }
+        }
+
+        [Fact]
+        public void AddCallbackFallingEdgeNotDetectedTest()
+        {
+            bool wasCalled = false;
+            using (GpioController controller = new GpioController(GetTestNumberingScheme(), GetTestDriver()))
+            {
+                controller.OpenPin(InputPin, PinMode.Input);
+                controller.OpenPin(OutputPin, PinMode.Output);
+                controller.Write(OutputPin, PinValue.Low);
+                controller.RegisterCallbackForPinValueChangedEvent(InputPin, PinEventTypes.Falling, callback);
+                controller.Write(OutputPin, PinValue.High);
+                Thread.Sleep(10);
```

Comment:
same here.  Should these values be hoisted into a static test variable that can be changed in one spot?

Reply:
sure that is better, will use a static variable